### PR TITLE
Release preparation script

### PR DESCRIPTION
**Pull request by achow101**
This is a bash script which does all of the automatable stuff for release preparation as specified by the release process doc.

### CODE REVIEW

**Comment by laanwj** [comment]
Nice concept!

I think we need it to be able to skip different steps though, e.g. if something was done recently already or we want to do it manually for some reason.

**Comment by achow101** [comment]
@laanwj I added a `--skip` option so you can skip one of the six steps. Also, I am not sure if this should include gen-manpages because that requires an already built version with the version number incremented.

File: release-prep.sh [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+# Help Message
+read -d '' usage <<- EOF
+Usage: $scriptName version block_chain_size
+
+Run this script from the Bitcoin Core Source root directory. This requires a current version of Bitcoin Core
+to be running at the time that this script is run.
+
+Arguments:
+version		    Version number to set following the MAJOR.MINOR.REVISION format. Only required if
```
Comment (kallewoof) [inline release-prep.sh]: Nit: Would avoid `\t`'s (or be consistent and not mix with space) if possible. (This looked weirdly indented in git.) Mostly seem to use tabs. Could maybe s&r `\t` -> ` `.

File: release-prep.sh [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+if [[ $chainparamsUpdate = true ]]
+then
+    # Update nMinimumChainWork and defaultAssumeValid
+    echo "Updating nMinimumChainWork and defaultAssumeValid"
+    blockchaininfo=`src/bitcoin-cli ${DATADIR} getblockchaininfo`
+    chainwork=`echo "$blockchaininfo" | jq -r '.chainwork'`
+    bestblockhash=`echo "$blockchaininfo" | jq -r '.bestblockhash'`
+    sed -i "0,/        consensus.nMinimumChainWork = uint256S(.*/s//        consensus.nMinimumChainWork = uint256S(\"0x$chainwork\");/" ./src/chainparams.cpp
+    sed -i "0,/        consensus.defaultAssumeValid = uint256S(.*/s//        consensus.defaultAssumeValid = uint256S(\"0x$bestblockhash\");/" ./src/chainparams.cpp
```
Comment (MarcoFalke) [inline release-prep.sh]: I think this is a manual process for each network and should probably not be automated.

**Comment by laanwj** [comment]
Going to merge this, being able to skip steps makes it easy enough to leave out steps that are not desirable to automate, or execute them on separate machines / at different times.

**Pull request merged**